src/Psalm/Internal/LanguageServer/Server/TextDocument.php

```diff
@@ -230,6 +230,28 @@ public function hover(TextDocumentIdentifier $textDocument, Position $position):
             return new Success(null);
         }
 
+        // Pretty print arrays (only).
+        if ( substr( $symbol_information['type'], 0, 6 ) === 'array{' ) {
+            $union = \Psalm\Type::parseString( $symbol_information['type'] );
+            $types = $union->getAtomicTypes();
+            if ( count( $types ) === 1 ) {
+                /** @var \Psalm\Type\Atomic\TKeyedArray */
+                $keyed_array = reset( $types );
+
+                $property_strings = array_map(
```

Comment:
You may want to use TKeyedArray::getId for that, it will handle escaping better and it will reduce the code here

Reply:
Oh, I read the code before the text in PR. This is needed for adding a line feed in there. But my comment on escaping still stand (also, did you quote every key on purpose? I believe Psalm can detect which key really need it).

Also, you may want to keep the usage of $keyed_array::KEY because it is sometimes replace by `callable-array`

Reply:
I tried `$keyed_array::KEY` but it seems that's interpreted as mixed value, I get `Right operand cannot be mixedPsalmhttps://psalm.dev/059` when trying that. I'm already checking `array{` is the sub-string of the type, so I think it won't actually be possible in this case.

Reply:
Added the escaping-- it looks like Psalm has no way to distinguish `array{ 0: true }` versus `array{ '0': true }` (I think PHP supports that?) so we don't need to handle this case for now.

Reply:
PHP doesn't support that, any numeric key is automatically casted to int in arrays